src/openassetio-core/include/openassetio/hostApi/EntityReferencePager.hpp

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2013-2023 The Foundry Visionmongers Ltd
+
+#pragma once
+
+#include <memory>
+
+#include <openassetio/EntityReference.hpp>
+#include <openassetio/typedefs.hpp>
+
+OPENASSETIO_FWD_DECLARE(managerApi, HostSession)
+OPENASSETIO_FWD_DECLARE(managerApi, EntityReferencePagerInterface)
+
+namespace openassetio {
+inline namespace OPENASSETIO_CORE_ABI_VERSION {
+namespace hostApi {
+
+OPENASSETIO_DECLARE_PTR(EntityReferencePager)
+
+/**
+ * The EntityReferencePager is the Host facing representation of a
+ * @fqref{managerApi.EntityReferencePagerInterface} implementation.
+ * The Manager class shouldn't be directly constructed by the host.
+ * The EntityReferencePager allows for the retrieval and traversal of large datasets
+ * in a paginated manner.
+ *
+ * Due to the variance of backends, construction, `hasNext`, `get` and
+ * `next` may all reasonably need to perform non trivial, networked
+ * operations, and thus performance characteristics should not be
+ * assumed.singleEntityReferencePager<Elem>
+ *
+ * Destruction of this object is a signal to the manager that the
+ * connection query is finished. For this reason you should avoid
+ * keeping hold of this object for longer than necessary.
+ */
+class OPENASSETIO_CORE_EXPORT EntityReferencePager {
+ public:
+  OPENASSETIO_ALIAS_PTR(EntityReferencePager)
+  using Page = EntityReferences;
+
+  [[nodiscard]] static EntityReferencePager::Ptr make(
+      managerApi::EntityReferencePagerInterfacePtr pagerInterface,
+      managerApi::HostSessionPtr hostSession);
```

Comment:
:thinking: This would mean we could no longer test it in isolation, so would take a little bit of work to refactor the C++ tests.

`friend`ing also means exposing _all_ `private` members.  Though I'm not against it in this case. 

I think if we're really limiting construction of `EntityReferencePager` objects to the `Manager`, then a better option would be to make `EntityReferencePager` a child class of the `Manager`.

Then `EntityReferencePagerInterface` should probably be a child class of `ManagerInterface` - but that's slightly ugly since manager plugin authors need to inherit from it.

I think there's an argument somewhere that probably applies, about loose coupling of components etc. So I'm happy to leave as-is too.

Will ponder some more.